src/analytics/patch/usePatchAnalytics.ts

```diff
@@ -55,9 +55,12 @@ interface P extends Properties {
 }
 interface PatchAnalytics extends Analytics<Action> {}
 
-export const usePatchAnalytics = (): PatchAnalytics => {
+export const usePatchAnalytics = (patchId?: string): PatchAnalytics => {
   const userId = useGetUserQuery();
-  const { id } = useParams<{ id: string }>();
+  const { id: idFromURL } = useParams<{ id: string }>();
+
+  const id = patchId || idFromURL;
```

Comment:
In this case i think it makes sense to not tie this hook to a specific implementation. It should not try to fetch a param from the url and it should instead always use a passed in param. So the implementer can pass in the appropriate id depending on the context it's implemented in.  (If its used on a patch page the implementer can pass in the appropriate id from the url. Or if its used in a card they can pass it in based off of that cards specific id.)